device_authorize_request_handler.go

```diff
@@ -0,0 +1,37 @@
+// Copyright © 2023 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package fosite
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/ory/fosite/i18n"
+	"github.com/ory/x/errorsx"
+)
+
+func (f *Fosite) NewDeviceAuthorizeRequest(ctx context.Context, r *http.Request) (DeviceAuthorizeRequester, error) {
+	request := NewDeviceAuthorizeRequest()
+	request.Lang = i18n.GetLangFromRequest(f.Config.GetMessageCatalog(ctx), r)
+
+	if err := r.ParseForm(); err != nil {
+		return nil, errorsx.WithStack(ErrInvalidRequest.WithHint("Unable to parse HTTP body, make sure to send a properly formatted form request body.").WithWrap(err).WithDebug(err.Error()))
+	}
+	request.Form = r.Form
+
+	verifier := request.GetRequestForm().Get("device_verifier")
```

Comment:
The `device_verifier` is something that is generated once the user has validated his `user_code`.

See here:
https://github.com/ory/hydra/blob/25346c6e7fd96ab59cb901c860f1eae36961de1c/consent/handler.go#L1126

It's important to understand that a choice has been made to go through hydra for the UserCode validation (instead of validating on the UI and then juste validating it)

Reply:
Thanks, makes sense!